###############################################################################
# Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
#

4076297 glXChooseVisual hangs if XInitThreads is called

diff -urp -x '*~' -x '*.orig' src/XlibInt.c src/XlibInt.c
--- src/XlibInt.c	2008-02-27 21:08:48.204605000 -0800
+++ src/XlibInt.c	2008-02-27 21:08:53.377329000 -0800
@@ -1339,7 +1339,8 @@ int _XRead(
        if (dpy->lock && dpy->lock->reply_bytes_left > 0)
        {
            dpy->lock->reply_bytes_left -= original_size;
-           if (dpy->lock->reply_bytes_left == 0) {
+           if (dpy->lock->reply_bytes_left <= 0) {
+	       dpy->lock->reply_bytes_left = 0;
 	       dpy->flags &= ~XlibDisplayReply;
                UnlockNextReplyReader(dpy);
 	   }
@@ -1564,7 +1565,8 @@ void _XReadPad(
        if (dpy->lock && dpy->lock->reply_bytes_left > 0)
        {
            dpy->lock->reply_bytes_left -= original_size;
-           if (dpy->lock->reply_bytes_left == 0) {
+           if (dpy->lock->reply_bytes_left <= 0) {
+	       dpy->lock->reply_bytes_left = 0;
 	       dpy->flags &= ~XlibDisplayReply;
                UnlockNextReplyReader(dpy);
 	   }
